Sources/ProjectDescription/Config.swift

```diff
@@ -35,6 +49,9 @@ public struct Config: Codable, Equatable {
         /// When passed, Tuist will enable code coverage for autogenerated default schemes according to given mode
         case enableCodeCoverage(CodeCoverageMode)
 
+        /// When passed, Tuist will customise the autogenerated test schemes according to given options
+        case testingOptions(TestingOptions)
+
```

Comment:
This is strictly related to `disableAutogeneratedSchemes`, so I think it shouldn't be a separate entry.
What if:
1. we rename `disableAutogeneratedSchemes` to `autogeneratedSchemes`
2. It takes an associated value enum which is either `.disabled` or `enabled(TestingOptions)`
3. To avoid a breaking change:
  - If nothing is specified, the default is the current one
  - We keep a static `disableSynthesizedResourceAccessors` which maps to `autogeneratedSchemes(.disabled)`

cc @fortmarek @pepicrft

Reply:
I agree!

Having `.autogeneratedSchemes(.disabled)` or `.autogeneratedSchemes(.enabled(testingOptions: [.parallelizable, .randomExecutionOrdering]))` would be better than splitting it into multiple root-level options.

> We keep a static disableSynthesizedResourceAccessors which maps to autogeneratedSchemes(.disabled)

I am not sure how `disableSynthesizedResourceAccessors` relates to `autogeneratedSchemes`?

Reply:
Oooops, I meant `disableAutogeneratedSchemes`, fixed

Reply:
makes sense, will update that + test default case match existing behaviour

Reply:
Have updated to this style